src/API.jl

```diff
@@ -1649,9 +1649,17 @@ function _activate_dep(dep_name::AbstractString)
         end
     end
 end
-function activate(path::AbstractString; shared::Bool=false, temp::Bool=false, io::IO=stderr_f())
+function activate(path::AbstractString; shared::Bool=false, temp::Bool=false, io::IO=stderr_f(), show_shared_hint::Bool=true)
     temp && pkgerror("Can not give `path` argument when creating a temporary environment")
     if !shared
+        if show_shared_hint && first(path) == '@'
```

Comment:
```suggestion
        if show_shared_hint && !isempty(path) &&  first(path) == '@'
```

`path` may be empty and thus `first(path)` will fail. See CI.